src/rules/checkLineAlignment.js

```diff
@@ -97,41 +103,93 @@ const checkNotAlignedPerTag = (utils, tag) => {
   utils.reportJSDoc('Expected JSDoc block lines to not be aligned.', tag, fix, true);
 };
 
-const checkAlignment = ({
+const filterApplicableTags = (jsdoc, applicableTags) => {
+  const filteredJsdoc = {
+    ...jsdoc,
+    source: jsdoc.source.filter((source) => {
+      if ('' === source.tokens.tag) {
+        return true;
+      }
+
+      return applicableTags.includes(source.tokens.tag.replace('@', ''));
+    }),
+    tags: jsdoc.tags.filter((tag) => {
+      return applicableTags.includes(tag.tag);
+    }),
+  };
+
+  return rewireSpecs(filteredJsdoc);
+};
+
+const getJsondocFormatted = ({
+  applicableTags,
   indent,
   jsdoc,
-  jsdocNode,
-  report,
   utils,
 }) => {
+  // It needs to be cloned. Otherwise, the transform overrides the original object.
+  const jsdocClone = cloneDeep(jsdoc);
   const transform = commentFlow(commentAlign(), commentIndent(indent.length));
-  const transformedJsdoc = transform(jsdoc);
+  const filteredClone = filterApplicableTags(jsdocClone, applicableTags);
+  const transformedJsdoc = transform(filteredClone);
+  const formatted = utils.stringify(transformedJsdoc);
+  const parsed = utils.commentParser({value: formatted}, indent, false);
 
-  const comment = '/*' + jsdocNode.value + '*/';
-  const formatted = utils.stringify(transformedJsdoc)
-    .trimStart();
+  return commentFlow(commentAlign())(parsed);
+};
 
-  if (comment !== formatted) {
-    report(
-      'Expected JSDoc block lines to be aligned.',
-      (fixer) => {
-        return fixer.replaceText(jsdocNode, formatted);
-      },
-    );
-  }
+const isTagSourcesEqual = (tag, otherTag) => {
+  return tag.source.every((source, index) => {
+    return source.source === otherTag.source[index].source;
+  });
+};
+
+const getFormattedSource = (tag, formattedTag) => {
+  return formattedTag.source.map((source, index) => {
+    return {
+      ...source,
+      number: tag.source[index].number,
+    };
+  });
+};
+
+const checkAlignment = ({
+  applicableTags,
+  foundTags,
+  indent,
+  jsdoc,
+  utils,
+}) => {
+  const jsdocFormatted = getJsondocFormatted({
+    applicableTags,
+    indent,
+    jsdoc,
+    utils,
+  });
+
+  foundTags.forEach((tag, index) => {
+    const formattedTag = jsdocFormatted.tags[index];
+    if (!isTagSourcesEqual(tag, formattedTag)) {
+      const fix = () => {
+        const formattedSource = getFormattedSource(tag, formattedTag);
+        utils.replaceTagSource(tag, formattedSource);
+      };
+      utils.reportJSDoc('Expected JSDoc block lines to be aligned.', tag, fix, true);
+    }
+  });
 };
 
 export default iterateJsdoc(({
   indent,
   jsdoc,
   jsdocNode,
-  report,
   context,
   utils,
 }) => {
   const {
     tags: applicableTags = ['param', 'arg', 'argument', 'property', 'prop', 'returns', 'return'],
   } = context.options[1] || {};
```

Comment:
This was pretty hard to read. A cleaner way of writing it would be:
```js
  const userTags = context.options?.[1]?.tags;
  const defaultTags = [
    'param',
    'arg',
    'argument',
    'property',
    'prop',
    'returns',
    'return'
  ];
  const applicableTags = userTags || defaultTags;
```
Or if this codebase doesn't use optional chaining:
```js
  const userTags = (
    context.options &&
    context.options[1] &&
    context.options[1].tags
  );
  const defaultTags = [
    'param',
    'arg',
    'argument',
    'property',
    'prop',
    'returns',
    'return'
  ];
  const applicableTags = userTags || defaultTags;
```

Reply:
Hey @TheJaredWilcurt !
Thank you for checking! Since it's not part of this PR, I'd prefer don't touch this part. If we want to refactor this part, maybe we could create a new PR for that. :)

Reply:
I know, I was just doin' [boy scout rule](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) ⚜️